### DISCUSSION TITLE
How do I exclude certain column(s) from feature extraction in tsfresh?

### DISCUSSION THREAD

**Discussion by kar10tik** [blue-yonder/tsfresh#1075] [Q&A]
I'm using tsfresh to extract features comprehensively from a timeseries. I wish to exclude certain columns in my dataset from feature extraction (i.e., absolute energy, min, max, etc. shouldn't be computed based on a list of certain columns I can specify). Is there any way I can do this in tsfresh without having to remove the specific column(s) from the DataFrame to be input to `tsfresh.feature_extraction.extraction.extract_features()` or drop features starting with the specific column name(s) after the feature extraction is complete?

**Comment by nils-braun** [comment]
Hi @kar10tik - there is the `kind_to_fc_parameters` parameter to the `extract_features` method. This will allow you to control explicitly which features to calculate for each column. So if you e.g. do not want to calculate a specific feature for a specific column, you can start with the default settings (https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_extraction.html#tsfresh.feature_extraction.settings.ComprehensiveFCParameters) and just remove the features you do not want. Like (pseudo-code)

```python
all_settings = ComprehensiveFCParameters()
reduced_setting = {key: value for key, value in all_settings.items() if key != "min"}
extract_features(..., kind_to_fc_parameters={"colA": all_settings, "colB": reuced_settings}
```
